### PR TITLE
chore: Cut `v2.10.1`

### PR DESCRIPTION
**Pull request by rexagod**
**What this PR does / why we need it**: `v2.10.1` release prep.

```markdown
## v2.10.1 / 2023-09-13

### Note
- This release addresses a regression introduced in https://github.com/kubernetes/kube-state-metrics/pull/2105.

* [BUGFIX] Remove FieldSelector from non-namespaced resources by @mrueg and @dgrisonnet in https://github.com/kubernetes/kube-state-metrics/pull/2190
```

***

### TODOs

- [ ] Release v2.10.1.
- [ ] `kpromo` changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,8 +77,8 @@ Generally, it is recommended to use the latest release of kube-state-metrics. If
 | **v2.7.0**         | v1.25                        |
 | **v2.8.2**         | v1.26                        |
 | **v2.9.2**         | v1.26                        |
-| **v2.10.0**        | v1.27                        |
-| **main**           | v1.27                        |
+| **v2.10.1**        | v1.27                        |
```
Comment (rexagod) [inline README.md:80]: (FYI) `main` needs to be updated to reflect the latest `2.10.x` version in the `README.md` so as to not cause conflicts when merging `2.10` back into `main`.

**Comment by rexagod** [comment]
Ah, let me update Go to `1.20.8` as well.

**Comment by rexagod** [comment]
I missed updating Go to `1.20.8` to address certain CVEs, on it.

**Comment by dgrisonnet** [comment]
@rexagod could you mention the go update in the changelog please?

**Pull request merged**